flood_adapt/object_model/interface/measures.py

```diff
@@ -174,70 +174,44 @@ def validate_type(cls, type: HazardType, values: Any) -> HazardType:
-    ) -> Optional[UnitfulLength]:
-        if values.get("type", "") == HazardType.total_storage:
-            if height is not None:
-                raise ValueError("Height cannot be set for total storage type measures")
-            return None  # Height is not required for total storage type measures
-        elif not isinstance(height, UnitfulLength):
-            raise ValueError("Height must be a UnitfulLength")
-        elif height.value <= 0:
-            raise ValueError("Height cannot be zero or negative")
-        return height
-
-    @validator("percent_area", always=True)
-    def validate_percent_area(
-        cls, percent_area: Optional[float], values: Any
-    ) -> Optional[float]:
-        if values.get("type", "") in [
-            HazardType.total_storage,
-            HazardType.water_square,
-        ]:
-            if percent_area is not None:
+    @model_validator(mode="after")
+    def validate_hazard_type_values(self) -> "GreenInfrastructureModel":
+        if self.type == HazardType.total_storage:
+            if self.height is not None or self.percent_area is not None:
                 raise ValueError(
-                    "Percent area cannot be set for total storage or water square type measures"
+                    "Height and percent_area cannot be set for total storage type measures"
                 )
-            return None  # Percent area is not required for total storage type measures
-        elif not isinstance(percent_area, float):
-            raise ValueError("Percent area must be a float")
-        elif percent_area < 0 or percent_area > 100:
-            raise ValueError("Percent area must be between 0 and 100")
-        return percent_area
-
-    @validator("aggregation_area_name", always=True)
-    def validate_aggregation_area_name(
-        cls, aggregation_area_name: Optional[str], values: Any
-    ) -> Optional[str]:
-        if (
-            values.get("selection_type", "") == SelectionType.aggregation_area
-            and aggregation_area_name is None
+            return self
+        elif self.type == HazardType.water_square:
+            if self.percent_area is not None:
+                raise ValueError(
+                    "Percentage_area cannot be set for water square type measures"
+                )
+            elif not isinstance(self.height, UnitfulHeight):
+                raise ValueError(
+                    "Height needs to be set for water square type measures"
+                )
+            return self
+        elif not isinstance(self.height, UnitfulHeight) or not isinstance(
+            self.percent_area, float
         ):
             raise ValueError(
-                "If `selection_type` is 'aggregation_area', then `aggregation_area_name` needs to be set."
+                "Height and percent_area needs to be set for greening type measures"
             )
-        return aggregation_area_name
+        return self
```

Comment:
Its not necessary, but I like explicitly using elifs for all cases and then throwing an error if the function loops through all options and doesn't find it. The field validator for type already does this check so its not required, but for readability I'd keep it.

Reply:
Technically this is already checked in the type validator but for readability, it might be better to do it in 1 statement indeed

Reply:
I mean do the explicit check for the hazard.Type here 
```
...
elif self.type == HazardType.greening:
    if not isinstance(self.height, UnitfulHeight) or not isinstance(self.percent_area, float):
         ....
else:
     raise ValueError( 'invalid/unsupported hazard type')
```

Reply:
I mean do the explicit check for the hazard.Type at line 195.
Then we have an elif for each hazard type and throw an error if we implement a new hazardtype but forgot to update this function.
```
...
elif self.type == HazardType.greening:
    if not isinstance(self.height, UnitfulHeight) or not isinstance(self.percent_area, float):
         ....
else:
     raise ValueError( 'invalid/unsupported hazard type')
```